Server/APIs/RoomsAPI.py

```diff
@@ -0,0 +1,110 @@
+from flask import Blueprint, request, jsonify
+from Model.Player import Player
+from Model.Room import Room
+from Model.RoomState import RoomState
+from Server.Database.DBMethods import DBMethods
+from Server.APIs.AuthenticationAPI import token_required
+import uuid
+from Common.RoomsAPIErrors import RoomsAPIErrors
+
+rooms_api = Blueprint('rooms_api', __name__)
+
+
+@rooms_api.route("/rooms", methods=['GET'])
+@token_required
+def get_rooms(current_player):
+    rooms_as_dicts = {}
+    index = 1
+    rooms_as_objects = get_available_rooms_as_objects()
+    for room_as_object in rooms_as_objects:
+        rooms_as_dicts[index] = room_as_object.to_dict()
+        index += 1
+    return jsonify(rooms_as_dicts)
+
+
+@rooms_api.route("/rooms/<room_name>", methods=['GET'])
+@token_required
+def get_room(current_player, room_name):
+    room_as_dict = DBMethods.get_room_details_from_db(room_name)
+    if room_as_dict is None:
+        room_as_dict = {}
+        return jsonify(room_as_dict)
+    else:
+        room_as_object = create_room_object_from_dict(room_as_dict)
+        return jsonify(room_as_object.to_dict())
+
+
+@rooms_api.route("/rooms", methods=['POST'])
+@token_required
+def host_room(current_player):
+    data = request.json
+    room_name = data['room_name']
+    if DBMethods.check_if_room_with_same_name_already_exists(room_name):
+        return RoomsAPIErrors.ROOM_NAME_OCCUPIED.value
+    if DBMethods.check_if_player_already_in_a_room(current_player.username):
+        return RoomsAPIErrors.PLAYER_ALREADY_IN_A_ROOM.value
+    room_id = str(uuid.uuid4())
+    room = Room(room_id, room_name, RoomState.WAITING.value, [current_player], current_player)
+    DBMethods.add_room_to_db(room)
+    DBMethods.update_player_linked_room_in_db(current_player.username, room_id)
+    return "Your room has been created successfully"
+
+
+@rooms_api.route("/rooms/<room_name>", methods=['PUT'])
+@token_required
+def join_room(current_player, room_name):
+    room_name_from_player_input = room_name
+    if DBMethods.check_if_player_already_in_a_room(current_player.username):
```

Comment:
In case DBMethods return only objects and not dict, your code is going to be even cleaner because you would not with dict at all!